### PR TITLE
CircleCI basic support

### PR DESCRIPTION
**Pull request by mattklein123**
This commit gives us basic IPv4 support via CircleCI. We still test IPv6 on
Travis as Circle does not yet support IPv4. This commit is going to fail the
master pushes. I need to make new keys and creds and configure env vars
in Circle. I would like to ship this and then I will work on that.

Signed-off-by: Matt Klein <mklein@lyft.com>

### CODE REVIEW

**Comment by mattklein123** [comment]
Note CircleCI doesn't appear to support IPv6, otherwise this is mostly working other than the push portions.

File: ci/do_circle_ci.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -e
+
+# bazel uses jgit internally and the default circle-ci .gitconfig says to
+# convert https://github.com to ssh://git@github.com, which jgit does not support.
+mv ~/.gitconfig ~/.gitconfig_save
+
+export ENVOY_SRCDIR="$(pwd)"
+export NUM_CPUS=8 # xlarge resource_class
```
Comment (htuch) [inline ci/do_circle_ci.sh]: Nit: why not let this be dynamically discovered to make it more robust?

File: ci/do_circle_ci.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ci/do_ci.sh $1
```
Comment (htuch) [inline ci/do_circle_ci.sh]: Nit: `"$1"`

File: ci/do_circle_ci.sh (hunk 1 above)
Comment (mattklein123) [reply ci/do_circle_ci.sh]: See the note here: https://circleci.com/docs/2.0/configuration-reference/#resource_class (It relates to how docker works unfortunately). I will add a comment here and generally clean this review up if we get IPv6 support.

Comment (htuch) [reply ci/do_circle_ci.sh]: As @alyssawilk suggests on other threads, if we want to keep Travis around for just some IPv6 stuff, then we don't need to block the Circle CI move.

**Comment by mattklein123** [comment]
@htuch updated

**Comment by mattklein123** [comment]
Actually let me add the build_images job also to this PR.

**Comment by htuch** [comment]
I don't see the Travis run in the above checks..

**Comment by mattklein123** [comment]
See continuous-integration/travis-ci/pr it's running.

**Pull request merged**